module/extension/Director/src/Director.hpp

```diff
@@ -373,20 +375,49 @@ namespace module::extension {
          */
         void run_task_pack(const TaskPack& pack);
 
+        /**
+         * A RunReasonLock instance will reset the current_run_reason back to OTHER_TRIGGER when it is destroyed.
+         *
+         * This is used so that the run reason can temporarily be set to something else, and then reset back to the
+         * default if there is an exception or other error.
+         */
+        using RunReasonLock = std::unique_ptr<void, std::function<void(void*)>>;
+
+        /**
+         * Holds a run reason as the current run reason for the current thread.
+         *
+         * It returns a lock object that acts as an RAII object that will reset the current run reason when it is
+         * destroyed.
+         *
+         * @param reason the reason to hold until the lock is destroyed
+         *
+         * @return a lock object that will reset the current run reason when it is destroyed
+         */
+        RunReasonLock hold_run_reason(const ::extension::behaviour::RunInfo::RunReason& reason);
+
     public:
         /// Called by the powerplant to build and setup the Director reactor.
         explicit Director(std::unique_ptr<NUClear::Environment> environment);
         virtual ~Director();
 
         /**
-         * @brief Provides the task data via the InformationSource interface so it can be accessed
+         * Provides the task data via the InformationSource interface so it can be accessed
          *
          * @param reaction_id the provider reaction that is requesting its information.
          *
          * @return the data that is stored in this reaction, or nullptr if it shouldn't be executing
          */
         std::shared_ptr<void> _get_task_data(const uint64_t& reaction_id) override;
 
+        /**
+         * Provides the RunInfo data via the InformationSource interface so it can be accessed
+         *
+         * @param reaction_id the provider reaction that is requesting its information.
+         *
+         * @return the information about why this task has been executed
+         */
+        ::extension::behaviour::RunInfo _get_run_info(const uint64_t& reaction_id) override;
```

Comment:
`_get_task_data`? `_get_run_info`? Have you been spending too much time in python?

Reply:
It's because in the information source there are static functions of the same name and they can't collide